roles/serviceassurance/tasks/component_alertmanager.yml

```diff
@@ -0,0 +1,62 @@
+- name: Set default Alertmanager manifest
+  set_fact:
+    alertmanager_manifest: |
+      apiVersion: monitoring.coreos.com/v1
+      kind: Alertmanager
+      metadata:
+        labels:
+          alertmanager: '{{ meta.name }}'
+        name: '{{ meta.name }}'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: 1
+        ruleSelector: {}
+        securityContext: {}
+        serviceAccountName: prometheus-k8s
+        serviceMonitorSelector:
+          matchLabels:
+            app: smart-gateway
+        storage:
+          class: {{ alertmanager_storage_class }}
+          resources: {{ alertmanager_storage_resources }}
+          selector: {{ alertmanager_storage_selector }}
+          volumeClaimTemplate:
+            spec:
+              resources:
+                requests:
+                  storage: {{ alertmanager_pvc_storage_request }}
+  when: alertmanager_manifest is not defined
+
+- name: Set default Alertmanager configuration manifest
+  set_fact:
+    alertmanager_config_manifest: |
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: alertmanager-'{{ meta.name }}'
+        namespace: '{{ meta.namespace }}'
+      type: Opaque
+      stringData:
+        alertmanager.yaml: |-
+          global:
+            resolve_timeout: 5m
+          route:
+            group_by: ['job']
+            group_wait: 30s
+            group_interval: 5m
+            repeat_interval: 12h
+            receiver: 'null'
+            routes:
+            - match:
+                alertname: DeadMansSwitch
```

Comment:
Does this Alert exist in our Prometheus config? I only see rules about ES that I think come from the ECL.

Reply:
I copy/pasted from tech-preview.

Alertmanager is only about sending alerts.

Reply:
> It was copy/pasted from tech-preview.

Okay, but AFAICT it's a routing rule for an alert that doesn't exist. Is that what we want here? We should not copy/paste things if they don't make sense. Does this make sense, or is it an oops?

> Alertmanager is only about sending alerts.

I don't know what you're trying to say?

Reply:
@csibbitt note that with the change to ECK that no rules are being populated at all by default. Can you capture the rules you're seeing from ECL? Maybe they will be useful to implement.

Reply:
For reference, here is a prometheus configuration for a semi-recent SAF deployment with no AM or PR:

```
global:
  scrape_interval: 30s
  scrape_timeout: 10s
  evaluation_interval: 30s
  external_labels:
    prometheus: sa-telemetry/saf-default
    prometheus_replica: prometheus-saf-default-0
alerting:
  alert_relabel_configs:
  - separator: ;
    regex: prometheus_replica
    replacement: $1
    action: labeldrop
rule_files:
- /etc/prometheus/rules/prometheus-saf-default-rulefiles-0/*.yaml
scrape_configs:
- job_name: sa-telemetry/saf-default/0
  scrape_interval: 1s
  scrape_timeout: 1s
  metrics_path: /metrics
  scheme: http
  kubernetes_sd_configs:
  - role: endpoints
    namespaces:
      names:
      - sa-telemetry
  relabel_configs:
  - source_labels: [__meta_kubernetes_service_label_app]
    separator: ;
    regex: smart-gateway
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_port_name]
    separator: ;
    regex: prom-http
    replacement: $1
    action: keep
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Node;(.*)
    target_label: node
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_endpoint_address_target_kind, __meta_kubernetes_endpoint_address_target_name]
    separator: ;
    regex: Pod;(.*)
    target_label: pod
    replacement: ${1}
    action: replace
  - source_labels: [__meta_kubernetes_namespace]
    separator: ;
    regex: (.*)
    target_label: namespace
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: service
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_pod_name]
    separator: ;
    regex: (.*)
    target_label: pod
    replacement: $1
    action: replace
  - source_labels: [__meta_kubernetes_service_name]
    separator: ;
    regex: (.*)
    target_label: job
    replacement: ${1}
    action: replace
  - separator: ;
    regex: (.*)
    target_label: endpoint
    replacement: prom-http
    action: replace
  metric_relabel_configs:
  - separator: ;
    regex: pod
    replacement: $1
    action: labeldrop
  - separator: ;
    regex: namespace
    replacement: $1
    action: labeldrop
  - separator: ;
    regex: instance
    replacement: $1
    action: labeldrop
  - separator: ;
    regex: job
    replacement: $1
    action: labeldrop
```

Reply:
Also this route is typically part of the example configurations for Alertmanager since it basically proves that alerts are working. It basically is always firing so that if your remote doesn't see it firing any more they can alert that alerts aren't working :)

I agree that we should really make sure we keep things as clean and intentional as possible here.

Reply:
Here are 3 basic objects for a single Alertmanager + Secret + PrometheusRules:

**Secret**
```
apiVersion: v1
kind: Secret
metadata:
  name: alertmanager-saf-default-test
  namespace: sa-telemetry
type: Opaque
stringData:
  alertmanager.yaml: |-
    global:
      resolve_timeout: "5m"
    receivers:
    - name: "null"
    route:
      group_by:
      - namespace
      group_interval: "5m"
      group_wait: "30s"
      receiver: "null"
      repeat_interval: "12h"
      routes:
      - match:
          alertname: "Watchdog"
        receiver: "null"
```

**Alertmanager**
```
apiVersion: monitoring.coreos.com/v1
kind: Alertmanager
metadata:
  creationTimestamp: null
  name: saf-default
  namespace: sa-telemetry
spec:
  replicas: 1
  securityContext: {}
```

**PrometheusRules**
```
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  labels:
    prometheus: saf-default
    role: alert-rules
  name: saf-default
  namespace: sa-telemetry
spec:
  groups:
    - name: general.rules
      rules:
        - alert: Watchdog
          annotations:
            message: >
              This is an alert meant to ensure that the entire alerting pipeline
              is functional.

              This alert is always firing, therefore it should always be firing
              in Alertmanager

              and always fire against a receiver. There are integrations with
              various notification

              mechanisms that send a notification when this alert is not firing.
              For example the

              "DeadMansSnitch" integration in PagerDuty.
          expr: vector(1)
          labels:
            severity: none
```

Reply:
I think the bare minimum here would be to to set the Secret to:

```
apiVersion: v1
kind: Secret
metadata:
  name: alertmanager-saf-default
  namespace: sa-telemetry
type: Opaque
stringData:
  alertmanager.yaml: |-
    route:
      receiver: dummy
    receivers:
      - name: dummy
```

This results in a configuration built in Alertmanager that looks like this (from the alertmanager webUI):

```
global:
  resolve_timeout: 5m
  http_config: {}
  smtp_hello: localhost
  smtp_require_tls: true
  pagerduty_url: https://events.pagerduty.com/v2/enqueue
  hipchat_api_url: https://api.hipchat.com/
  opsgenie_api_url: https://api.opsgenie.com/
  wechat_api_url: https://qyapi.weixin.qq.com/cgi-bin/
  victorops_api_url: https://alert.victorops.com/integrations/generic/20131114/alert/
route:
  receiver: dummy
receivers:
- name: dummy
templates: []
```

And then set the `Alertmanager` object as above, and you'll have a loaded Alertmanager. Don't load any PrometheusRules at this point as there will be no receiver for them.

Reply:
Also in the templates above, `s/saf-default/{{ meta.name }}/g`